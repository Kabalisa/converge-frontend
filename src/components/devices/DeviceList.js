import React from 'react';
import '../../assets/styles/devicelist.scss';
import ColGroup from '../helpers/ColGroup';
import TableHead from '../helpers/TableHead';
import Device from './Device';
import devices from '../../fixtures/devices';
import MenuTitle from '../commons/MenuTitle';

const deviceListItems = devices.map(device => (
  <Device device={device} key={device.name} />
));

/**
 * Device List Component
 *
 * @returns {JSX}
 */
const DeviceList = () => (
  <div className="settings-devices-list">
    <div className="settings-devices-control">
      <MenuTitle title="Devices" />
    </div>
    <table>
      <ColGroup />
      <TableHead
        titles={['Name', 'Type', 'Date Added', 'Last Seen', 'Location']}
      />
      <tbody>{deviceListItems}</tbody>
    </table>
  </div>
);

export default DeviceList;
