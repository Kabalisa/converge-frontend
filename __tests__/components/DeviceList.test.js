import React from 'react';
import { shallow } from 'enzyme';
import DeviceList from '../../src/components/DeviceList';

describe('DeviceList Component', () => {
  it('renders correctly from memory', () => {
    expect(shallow(<DeviceList />)).toMatchSnapshot();
  });
});
