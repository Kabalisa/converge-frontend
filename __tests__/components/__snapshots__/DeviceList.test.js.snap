// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviceList Component renders correctly from memory 1`] = `
<div
  className="settings-devices-list"
>
  <table>
    <ColGroup
      numberOfMiddleColumns={2}
    />
    <TableHead
      titles={
        Array [
          "Name",
          "Type",
          "Date Added",
          "Last Seen",
          "Location",
        ]
      }
    />
    <tbody>
      <Device
        device={
          Object {
            "date": "14 Aug 2018",
            "last": "16 Feb 2018",
            "location": "Nairobi",
            "name": "Samsung tablet",
            "type": "Internal Display",
          }
        }
        key="Samsung tablet"
      />
      <Device
        device={
          Object {
            "date": "14 Aug 2018",
            "last": "16 Feb 2018",
            "location": "Nairobi",
            "name": "Samsung Galaxy",
            "type": "Internal Display",
          }
        }
        key="Samsung Galaxy"
      />
      <Device
        device={
          Object {
            "date": "14 Aug 2018",
            "last": "16 Feb 2018",
            "location": "Nairobi",
            "name": "Samsung duos",
            "type": "Internal Display",
          }
        }
        key="Samsung duos"
      />
      <Device
        device={
          Object {
            "date": "14 Aug 2018",
            "last": "16 Feb 2018",
            "location": "Nairobi",
            "name": "Samsung duos",
            "type": "Internal Display",
          }
        }
        key="Samsung duos"
      />
    </tbody>
  </table>
</div>
`;
